source/volume_of_fluid/assembler.cc

```diff
@@ -0,0 +1,667 @@
+                               scratch.face_finite_element_values.JxW(q);
+
+                  boundary_fluid_flux += this->get_boundary_composition_manager().boundary_composition(
+                                           face->boundary_id(),
+                                           scratch.face_finite_element_values.quadrature_point(q),
+                                           field.composition_index) *
+                                         current_u *
+                                         scratch.face_finite_element_values.normal_vector(q) *
+                                         scratch.face_finite_element_values.JxW(q);
+
+                }
+            }
+
+          // Due to inability to reference this cell's values at the interface,
+          // need to do explicit calculation
+          if (f_dir_pos)
+            {
+              face_ls_d = cell_i_d - 0.5*cell_i_normal[f_dim];
+              face_ls_time_grad = (face_flux/cell_vol)*cell_i_normal[f_dim];
+            }
+          else
+            {
+              face_ls_d = cell_i_d + 0.5*cell_i_normal[f_dim];
+              face_ls_time_grad = -(face_flux/cell_vol)*cell_i_normal[f_dim];
+            }
+
+          // Calculate outward flux
+          double flux_volume_of_fluid;
+          if (std::abs(face_flux) < volume_fraction_threshold*cell_vol)
+            {
+              flux_volume_of_fluid = cell_volume_of_fluid;
+            }
+          else if (face_flux < 0.0) // edge is upwind, currently assume zero inflow
+            {
+              flux_volume_of_fluid = boundary_fluid_flux/face_flux;
+            }
+          else // Cell is upwind, outflow boundary
+            {
+              flux_volume_of_fluid = VolumeOfFluid::calc_volume_of_fluid_flux_edge<dim> (f_dim,
+                                                                                         face_ls_time_grad,
+                                                                                         cell_i_normal,
+                                                                                         face_ls_d);
+            }
+
+          // Add fluxes to RHS
+          data.local_rhs[0] -= (flux_volume_of_fluid-cell_volume_of_fluid) * face_flux;
+        }
+    }
+
+    template <int dim>
+    void VolumeOfFluidAssembler<dim>::local_assemble_internal_face_volume_of_fluid_system (const VolumeOfFluidField<dim> field,
+        const unsigned int calc_dir,
+        bool update_from_old,
+        const typename DoFHandler<dim>::active_cell_iterator &cell,
+        const unsigned int face_no,
+        internal::Assembly::Scratch::VolumeOfFluidSystem<dim> &scratch,
+        internal::Assembly::CopyData::VolumeOfFluidSystem<dim> &data) const
+    {
+      const bool old_velocity_avail = (this->get_timestep_number() > 0);
+
+      const unsigned int f_dim = face_no/2; // Obtain dimension
+      const bool f_dir_pos = (face_no%2==1);
```

Comment:
Please extend the comment, and find a better name for this variable. What is `f_dir_pos`?